src/ng/urlUtils.js

```diff
@@ -91,6 +91,59 @@ function urlResolve(url) {
  */
 function urlIsSameOrigin(requestUrl) {
   var parsed = (isString(requestUrl)) ? urlResolve(requestUrl) : requestUrl;
-  return (parsed.protocol === originUrl.protocol &&
-          parsed.host === originUrl.host);
+  return urlsAreSameOrigin(parsed, originUrl);
+}
+
+/**
+ * Parse a request URL and determine whether it is same-origin as the current document base URL.
+ *
+ * Note: The base URL is usually the same the document location (`location.href`) but can
+ * overriden by using the `<base>` tag.
+ *
+ * @param {string|object} requestUrl The url of the request as a string that will be resolved
+ * or a parsed URL object.
+ * @returns {boolean} Whether the URL is same-origin as the document base URL.
+ */
+function urlIsSameOriginAsBaseUrl(requestUrl) {
+  var parsed = (isString(requestUrl)) ? urlResolve(requestUrl) : requestUrl;
+  var base = urlResolve(getBaseUrl());
+  return urlsAreSameOrigin(parsed, base);
+}
+
+/**
+ * Determines if two URLs share the same origin.
+ *
+ * @param {object} url1 First URL to compare. Must be a normalized URL in the form of a
+ *     dictionary object returned by `urlResolve()`.
+ * @param {object} url2 Second URL to compare. Must be a normalized URL in the form of a
+ *     dictionary object returned by `urlResolve()`.
+ * @return {boolean} True if both URLs have the same origin, and false otherwise.
+ */
+function urlsAreSameOrigin(url1, url2) {
+  // IE sometimes includes a port in the 'host' property, even if it is the default 80 port so
```

Comment:
Why is that a problem? Both URLs should be consistent (either contain the port or don't).
Under what circumstances can the URLs differ in that regard?